eq-author/src/components/PositionModal/index.js

```diff
@@ -32,64 +32,117 @@ const Trigger = styled.button.attrs({ type: "button" })`
   }
 `;
 
+const Indent = styled(Option)`
+  margin-left: ${({ indent }) => (indent ? 1 : 0)}em;
+`;
+
 const PositionModal = ({ options, onMove, selected }) => {
-  const previousPosition = useMemo(
-    () => options.findIndex(({ id }) => id === selected.id),
-    [options, selected]
-  );
+  const positionButtonId = uniqueId("PositionModal");
   const [isOpen, setIsOpen] = useState(false);
-  const [selectedPosition, setSelectedPosition] = useState(previousPosition);
-  const [positionButtonId] = useState(uniqueId("PositionModal"));
+  const previousIndex = options.findIndex(({ id }) => id === selected.id);
+  const previousPosition = useRef(previousIndex > -1 ? previousIndex : 0);
+  const [{ position, item }, setOption] = useState({
+    position: previousPosition.current,
+    item: options[previousPosition.current],
+  });
+
+  useEffect(() => {
+    // resets the position of the selected item when changing sections
+    const previousIndex = options.findIndex(({ id }) => id === selected.id);
+    previousPosition.current = previousIndex > -1 ? previousIndex : 0;
+    setOption((prev) => ({ ...prev, position: previousPosition.current }));
+  }, [options, selected]);
 
   const orderedOptions = options.filter(({ id }) => id !== selected.id);
-  orderedOptions.splice(selectedPosition, 0, selected);
+  selected.parentEnabled = item?.parentEnabled;
+  orderedOptions.splice(position, 0, selected);
 
   const handleClose = () => {
     setIsOpen(false);
-    setSelectedPosition(previousPosition);
+    setOption({
+      position: previousPosition.current,
+      item: orderedOptions[previousPosition.current],
+    });
   };
 
-  const handleOpen = () => setIsOpen(true);
-
-  const handleChange = ({ value }) => setSelectedPosition(parseInt(value, 10));
+  const handleChange = ({ value }) => {
+    const count =
+      orderedOptions[value].__typename === "Folder" && value - position >= 0
+        ? orderedOptions.filter(
+            ({ parentId }) => parentId === orderedOptions[value].id
+          ).length
+        : 0;
```

Comment:
You could split this up a bit to improve readability.

```suggestion
    const filteredOptions = orderedOptions.filter(
            ({ parentId }) => parentId === orderedOptions[value].id
          );
          
    const count =
      orderedOptions[value].__typename === "Folder" && value - position >= 0
        ? filteredOptions.length : 0;
```